### PR TITLE
Import core bundle

### PR DESCRIPTION
**Pull request by core23**
## Subject

I am targeting this branch, because there is no release yet.

## Changelog

```markdown
### Added
- Added all classes from `sonata-project/core-bundle`
```

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -21,10 +21,26 @@
         }
     ],
     "require": {
-        "php": "^7.1"
+        "php": "^7.1",
```
Comment (OskarStark) [inline composer.json]: I think we should go to 7.2

Comment (core23) [reply composer.json]: This is just an import. Improvements in code should be done afterwards

Comment (OskarStark) [reply composer.json]: Ok but before the first release

Comment (OskarStark) [reply composer.json]: Ok but before the first release

File: docs/reference/flash_messages.rst [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    $flashManager = $this->get('sonata.twig.flashmessage.manager');
+
+    $messages = $flashManager->get('success');
+
+To use this feature in your templates, include the following template:
+
+.. code-block:: jinja
+
+    {% include '@SonataCore/FlashMessage/render.html.twig' %}
```
Comment (greg0ire) [inline docs/reference/flash_messages.rst]: This should probably change, right?

File: src/Extension/FlashMessageExtension.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\Twig\Extension;
+
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+/**
+ * This is the Sonata core flash message Twig extension.
```
Comment (greg0ire) [inline src/Extension/FlashMessageExtension.php]:
```suggestion
 * This is the Sonata flash message Twig extension.
```

File: src/Extension/FlashMessageRuntime.php [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\Twig\Extension;
+
+use Sonata\Twig\FlashMessage\FlashManager;
+
+/**
+ * This is the Sonata core flash message Twig runtime.
```
Comment (greg0ire) [inline src/Extension/FlashMessageRuntime.php]:
```suggestion
 * This is the Sonata flash message Twig runtime.
```

File: src/Extension/FlashMessageExtension.php (hunk 3 above)
Comment (greg0ire) [reply src/Extension/FlashMessageExtension.php]: Or maybe just remove that comment entirely :P

File: src/Extension/FlashMessageRuntime.php [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+     */
+    private $flashManager;
+
+    public function __construct(FlashManager $flashManager)
+    {
+        $this->flashManager = $flashManager;
+    }
+
+    /**
+     * Returns flash messages handled by Sonata core flash manager.
```
Comment (greg0ire) [inline src/Extension/FlashMessageRuntime.php]:
```suggestion
     * Returns flash messages handled by Sonata flash manager.
```

File: src/Extension/FlashMessageRuntime.php [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+     *
+     * @param string $type Type of flash message
+     */
+    public function getFlashMessages(string $type): array
+    {
+        return $this->flashManager->get($type);
+    }
+
+    /**
+     * Returns flash messages types handled by Sonata core flash manager.
```
Comment (greg0ire) [inline src/Extension/FlashMessageRuntime.php]:
```suggestion
     * Returns flash messages types handled by Sonata flash manager.
```

File: src/FlashMessage/FlashManager.php [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+     */
+    private $types;
+
+    /**
+     * @var array
+     */
+    private $cssClasses;
+
+    /**
+     * @param array $types      Sonata core types array (defined in configuration)
```
Comment (greg0ire) [inline src/FlashMessage/FlashManager.php]:
```suggestion
     * @param array $types      Sonata types array (defined in configuration)
```

File: src/FlashMessage/FlashManager.php [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+
+    public function getStatusClass($object, ?string $statusName = null, string $default = ''): string
+    {
+        return \array_key_exists($object, $this->cssClasses)
+            ? $this->cssClasses[$object]
+            : $default;
+    }
+
+    /**
+     * Returns Sonata core flash message types.
```
Comment (greg0ire) [inline src/FlashMessage/FlashManager.php]:
```suggestion
     * Returns Sonata flash message types.
```

File: src/FlashMessage/FlashManager.php [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+    /**
+     * Returns Symfony session service.
+     */
+    public function getSession(): SessionInterface
+    {
+        return $this->session;
+    }
+
+    /**
+     * Returns flash bag messages for correct type after renaming with Sonata core type.
```
Comment (greg0ire) [inline src/FlashMessage/FlashManager.php]:
```suggestion
     * Returns flash bag messages for correct type after renaming with Sonata type.
```

File: src/FlashMessage/FlashManager.php [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+            foreach ($values as $value => $options) {
+                $this->rename($type, $value);
+            }
+        }
+    }
+
+    /**
+     * Process flash message type rename.
+     *
+     * @param string $type  Sonata core flash message type
```
Comment (greg0ire) [inline src/FlashMessage/FlashManager.php]:
```suggestion
     * @param string $type  Sonata flash message type
```

File: tests/FlashMessage/FlashManagerTest.php [hunk 11]
```diff
@@ -0,0 +1,221 @@
...
+     *
+     * @return \Symfony\Component\HttpFoundation\Session\Session
+     */
+    protected function getSession()
+    {
+        return new Session(new MockArraySessionStorage(), new AttributeBag(), new FlashBag());
+    }
+
+    /**
+     * Returns Sonata core flash manager.
```
Comment (greg0ire) [inline tests/FlashMessage/FlashManagerTest.php]:
```suggestion
     * Returns Sonata flash manager.
```

File: src/Extension/FlashMessageExtension.php (hunk 3 above)
Comment (greg0ire) [reply src/Extension/FlashMessageExtension.php]: Same for other comments below

**Pull request merged**